src/Components/WebAssembly/WebAssembly/src/Hosting/IWebAssemblyHostEnvironment.cs

```diff
@@ -13,5 +15,10 @@ public interface IWebAssemblyHostEnvironment
         /// Configured to "Production" when not specified by the host.
         /// </summary>
         string Environment { get; }
+
+        /// <summary>
+        /// Gets the applications base adddres.
+        /// </summary>
+        public Uri BaseAddress { get;  }
```

Comment:
Instead of using `System.Uri`, could we use a `string`? That brings this into consistency with `NavigationManager`, which exposes `string BaseUri { get; }`, and the general pattern in ASP.NET Core which is to prefer using strings to represent URIs.

Reply:
Seems sensible. Can do.

Reply:
For some general context on this - `System.Uri` does some messy things with canonicalization of URLS: https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.8#remarks

Since we are the platform we generally make sure that's done correctly at another layer of the system.

Reply:
![image](https://user-images.githubusercontent.com/1857993/77194468-45b32700-6a9d-11ea-8890-4db1d6e75d54.png)